src/generators/AbstractGenerator.ts

```diff
@@ -63,6 +65,16 @@ export interface AbstractGeneratorRenderCompleteModelArgs<
   options?: DeepPartial<Options>;
 }
 
+export interface InputProcessorOptions {
+  type?: InputType
+}
+
+export enum InputType {
+  FILE = 'file',
+  DOCUMENT = 'document',
+  OBJECT = 'object'
```

Comment:
I actually dont think we need this specifically 🤔 The idea is good though!

`file` can be detected https://en.wikipedia.org/wiki/File_URI_scheme
`document` can also be detected most of the times https://github.com/asyncapi/modelina/blob/e53c2af1095e020667ee2a14d4f0aa09f5546796/src/processors/AsyncAPIInputProcessor.ts#L393
`object` can also be detected with typeof and the above check.

I think it makes more sense to have `ProcessorType` where you force the input to be loaded as a specific input processor.

Reply:
@jonaslagoni , I was hoping to make input as file path and not the URI. I am not aware of whole code so can you tell me if we could have other strings as input. Or can I assume that if there is a string as input then it will be a file path.
- If every string is a file path then we can easily check if the file exists and throw error if not.
- If every string is not a file path we have to check if file exists and if not we have to pass it further to the processor without throwing error.

Reply:
Also can you tell me why is the input type is `any`? Cant we make it dynamic to use type effectively?

Reply:
So this all comes down to how we want to process inputs, check out if you have not yet https://github.com/asyncapi/modelina/blob/master/docs/input-processing.md

The reason why we have any is because we support such a wide range of inputs, i.e. Stringified objects, already parsed inputs, [typescript file](https://modelina.org/examples?selectedExample=java-from-typescript-type), etc.

Enabling support for file input has to follow a similar strategy as the one we are using. The reason I suggested URI as the format, is because its super simple to check it for the resolvement strategy that should be used, i.e. for local files you check for `file://` and if it's HTTP for example, we can add support for that as well just by looking at that first part.

So, the question is, how can we integrate file inputs into our current architecture?

My suggestion is that we slightly alter https://github.com/asyncapi/modelina/blob/e53c2af1095e020667ee2a14d4f0aa09f5546796/src/processors/InputProcessor.ts#L48

To check which type of input was provided. Then based on that it will move that input down to the right input processor 🙂 I.e. if it's local file input we can pass it down to the AsyncAPI processor (at least for now) either through the same `process` function or a new one called `processFile` (maybe).

Does that make sense?

Reply:
Yes. It makes sense to me. I have updated the code to handle file URI. As a future improvement I suggest we keep the input type dynamic and let Input processor decide its type. This will give a good Intelephense to the end user. 😊